sfputil/debug.py

```diff
@@ -0,0 +1,327 @@
+
+
+def get_subport_lane_mask(subport, lane_count):
+    """
+    Get the lane mask for the given subport and lane count.
+
+    This method calculates the lane mask based on the subport and lane count.
+
+    Args:
+        subport (int): The subport number to calculate the lane mask for.
+        lane_count (int): The number of lanes per subport.
+
+    Returns:
+        int: The lane mask calculated for the given subport and lane count.
+    """
+    # Calculating the lane mask using bitwise operations.
+    return ((1 << lane_count) - 1) << ((subport - 1) * lane_count)
+
+
+def get_sfp_object(port_name):
+    """
+    Retrieve the SFP object for a given port.
+
+    This function checks whether the port is a valid RJ45 port or if an SFP is present.
+    If valid, it retrieves the SFP object for further operations.
+
+    Args:
+        port_name (str): The name of the logical port to fetch the SFP object for.
+
+    Returns:
+        SfpBase: The SFP object associated with the port.
+
+    Raises:
+        SystemExit: If the port is an RJ45 or the SFP EEPROM is not present.
+    """
+    # Retrieve the physical port corresponding to the logical port.
+    physical_port = logical_port_to_physical_port_index(port_name)
+    # Fetch the SFP object for the physical port.
+    sfp = platform_chassis.get_sfp(physical_port)
+
+    # Check if the port is an RJ45 port and exit if so.
+    if is_rj45_port(port_name):
+        click.echo(f"{port_name}: This functionality is not applicable for RJ45 port")
+        sys.exit(EXIT_FAIL)
+
+    # Check if the SFP EEPROM is present and exit if not.
+    if not is_sfp_present(port_name):
+        click.echo(f"{port_name}: SFP EEPROM not detected")
+        sys.exit(EXIT_FAIL)
+
+    return sfp
+
+
+def get_subport(port_name):
+    """
+    Retrieve subport from the CONFIG_DB.
+
+    This function fetches the subport value from the configuration database. If no subport is specified, it defaults to 1.
+
+    Args:
+        port_name (str): The logical port name to retrieve the subport for.
+
+    Returns:
+        int: The subport associated with the port (default is 1 if not found).
+
+    Raises:
+        SystemExit: If the subport value is not found in the CONFIG_DB or there is a failure connecting to it.
+    """
+    # Get the namespace for the port
+    namespace = multi_asic.get_namespace_for_port(port_name)
+    # Connect to the CONFIG_DB to fetch subport information
+    config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
+
+    if config_db is not None:
+        config_db.connect()
+        try:
+            # Try to fetch the subport for the given port.
+            subport = int(config_db.get(config_db.CONFIG_DB, f'PORT|{port_name}', 'subport'))
+        except TypeError:
+            # If no subport value is found, exit with an error.
+            click.echo(f"{port_name}: subport is not present in CONFIG_DB")
+            sys.exit(EXIT_FAIL)
```

Comment:
@vdahiya12 Can we create a function to retrieve value from DB for a given field? This way, we can avoid code duplication in the loopback function.

Reply:
fixed